test/suite/test_verbose02.py

```diff
@@ -0,0 +1,119 @@
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+import wiredtiger, wttest
+from test_verbose01 import test_verbose_base
+
+# test_verbose02.py
+# Verify basic uses of the verbose configuration API when categories and valid/invalid verbosity
+# levels are specified.
+class test_verbose02(test_verbose_base):
+    collection_cfg = 'key_format=S,value_format=S'
+
+    # Test use cases passing single verbose categories, ensuring we only produce verbose output for
+    # the single category.
+    def test_verbose_single(self):
+        # Close the initial connection. We will be opening new connections with different verbosity
+        # settings throughout this test.
+        self.close_conn()
+
+        # Test passing a single verbose category, 'api' along with the verbosity level
+        # WT_VERBOSE_DEBUG (1). Ensuring the only verbose output generated is related to the 'api'
+        # category.
+        with self.expect_verbose(['api:1'], ['WT_VERB_API']) as conn:
+            # Perform a set of simple API operations to generate verbose API messages.
+            uri = 'table:test_verbose01_api'
+            session = conn.open_session()
+            session.create(uri, self.collection_cfg)
+            c = session.open_cursor(uri)
+            c['api'] = 'api'
+            c.close()
+            session.close()
+
+        # At this time, there is no verbose messages with the category WT_VERB_API and the verbosity
+        # level WT_VERBOSE_INFO (0), hence we don't expect any output.
+        with self.expect_verbose(['api:0'], ['WT_VERB_API'], False) as conn:
+            uri = 'table:test_verbose01_api'
+            session = conn.open_session()
+            session.create(uri, self.collection_cfg)
+            c = session.open_cursor(uri)
+            c['api'] = 'api'
+            c.close()
+            session.close()
+
+        # Test passing another single verbose category, 'compact' with different verbosity levels.
+        # Since there are verbose message with the category WT_VERB_COMPACT and the verbosity levels
+        # WT_VERBOSE_INFO (0) and WT_VERBOSE_DEBUG (1), we can test them both.
```

Comment:
I assume that setting verbosity to level 1 should also print level 0 traces. If so are we able to check this?
Similarly when we pass in a category without a level (`['api']`) we can check this prints all verbosity levels

Reply:
We will be able to check that the expected level is printed soon. It will be done as part of WT-8260.

Example of current traces when the `api` category is enabled:
```
[1636325735:250078][32792:0x7f26afea5700], test_checkpoint01.test_checkpoint_cursor.test_checkpoint_dne(table), file:WiredTigerHS.wt, WT_CURSOR.reset: [WT_VERB_API] CALL: WT_CURSOR:reset
```

We only see the category `WT_VERB_API` which is checked by the `expect_verbose` function. The level is not being printed yet, hence we cannot check it for now.